File: backend/domain/models/box_edge.go
package models

import (
	"github.com/google/uuid"
	"gorm.io/gorm"
)

type BoxEdge struct {
	ID          uuid.UUID `gorm:"type:uuid;primaryKey"`
	ParentBoxID uuid.UUID `gorm:"type:uuid;not null;index"`
	ChildBoxID  uuid.UUID `gorm:"type:uuid;not null;index"`
	Position    int       `gorm:"not null;default:0"`
	ParentBox   *Box      `gorm:"foreignKey:ParentBoxID"`
	ChildBox    *Box      `gorm:"foreignKey:ChildBoxID"`
}

func (e *BoxEdge) BeforeCreate(tx *gorm.DB) error {
	if e.ID == uuid.Nil {
		e.ID = uuid.New()
	}
	return nil
}

File: src/Domain/Entities/BoxEdge.cs
namespace CuRoadmap.Domain.Entities;

public class BoxEdge
{
    public Guid Id { get; set; }
    public Guid ParentBoxId { get; set; }
    public Guid ChildBoxId { get; set; }
    public int Position { get; set; }

    public Box ParentBox { get; set; } = null!;
    public Box ChildBox { get; set; } = null!;
}
